src/index/base.cpp

```diff
@@ -60,33 +60,34 @@ bool BaseIndex::Init()
     }
 
     LOCK(cs_main);
+    CChain& active_chain = m_chainstate->m_chain;
```

Comment:
I really don't like `CChain` becoming part of chainstate's public interface. It's too low level - code outside validation shouldn't be interacting directly with that object and should be using higher level interface functions.

Reply:
Do you mean the `m_chain` member specifically or any kind of `CChain`? It seems that we were already using a `CChain` previously when we were caling `::ChainActive()`. Happy to document and add to #21766 so nothing gets lost!